grafana/grafana
# Elasticsearch: remove already-disabled code

Fixes https://github.com/grafana/grafana/issues/48741

in Grafana9.0.0 we removed support for elasticsearch versions after their end-of-life (see https://github.com/grafana/grafana/pull/48715 ). 

this allows us to cleanup&simplifiy some code, which happens in this PR.

NOTE: i focused on just doing the cleanup/code-removal and nothing else. one thing we should do afterwards is: currently when you create a new elastic datasource, it defaults to `5.0.0`. it should default to `8.0.0`)


how to test:
- there are no specific features we should test with, in general use the elastic-search datasource a little and see if anything breaks.
- try it with both elastic7 and elastic8, this is how:
    - `make devenv sources=elastic` .. this runs with elastic-8.2.0
    - `make devenv sources=elastic elastic_version=7.12.1` ... this runs with elastic-7.12.1
